edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/SupportBundleRequest.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Requests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Newtonsoft.Json;
+
+    public class SupportBundleRequest
+    {
+        public SupportBundleRequest(string schemaVersion, string sasUrl, string since, string iothubHostname, bool? edgeRuntimeOnly)
+        {
+            this.SchemaVersion = Preconditions.CheckNonWhiteSpace(schemaVersion, nameof(schemaVersion));
+            this.SasUrl = Preconditions.CheckNotNull(sasUrl, nameof(sasUrl));
+            this.Since = Option.Maybe(since);
+            this.IothubHostname = Option.Maybe(iothubHostname);
+            this.EdgeRuntimeOnly = Option.Maybe(edgeRuntimeOnly);
+        }
+
+        [JsonProperty("schemaVersion")]
+        public string SchemaVersion { get; }
+
+        [JsonIgnore]
+        public string SasUrl { get; }
+
+        [JsonProperty("since")]
+        public Option<string> Since { get; }
+
+        [JsonProperty("iothubHostname")]
+        public Option<string> IothubHostname { get; }
```

Comment:
How is this used?

Reply:
iotedge check has it as an option, so the support bundle does too

Reply:
What happens if a user specifies a different IoTHub? 
I think this parameter definitely seems odd - I would say remove it and use the IOTHub that EdgeAgent is using. 
@veyalla ?

Reply:
Using IoT Hub that edge agent is using would actually be an improvement over running support-bundle locally because currently we cannot run any connectivity tests when using DPS because we can't get the iot hub name from config yaml. 

However, it would cause inconsistent behavior for the same command when run locally vs remotely which is weird. There was some thoughts from @arsing around getting the iot hub name from a cached provisioning result when run locally on a DPS provisioned device. If that can be implemented we could have a consistent experience.

Reply:
`provisioning_backup.json` has the hub name, so `iotedge check` / `support-bundle` could read it from there.

Reply:
(If you do make that change, you'll probably want to keep the `--iothub-hostname` parameter to `check` / `support-bundle` so that passing it is not an error, such as when running old EA with new host package.)

Reply:
IMO, we should make the change Arnav is suggesting, it will improve troubleshooting experience for customers running check / support-bundle locally and also provide consistency when running it locally or remotely. Thoughts @CindyXing ?

Reply:
Yes, wherever we can infer the IoThub, we should I can't think of a scenario where support bundle or iotedge check would need to talk to a different IoTHub.

Reply:
So for this pr, I'll remove the iothubHostname param, and just fill it in from ea's known hostname